kubeflow/trainer/backends/localprocess/utils.py

```diff
@@ -0,0 +1,302 @@
+        trainer_packages=trainer_packages,
+    )
+
+    options = [f"--index-url {pip_index_urls[0]}"]
+    options.extend(f"--extra-index-url {extra_index_url}" for extra_index_url in pip_index_urls[1:])
+
+    """
+           PIP_DISABLE_PIP_VERSION_CHECK=1 pip install $QUIET $AS_USER \
+       --no-warn-script-location $PIP_INDEX $PACKAGE_STR
+       """
+    mapping = {
+        "QUIET": "--quiet" if quiet else "",
+        "PIP_INDEX": " ".join(options),
+        "PACKAGE_STR": '"{}"'.format('" "'.join(packages)),  # quote deps
+    }
+    t = Template(local_exec_constants.DEPENDENCIES_SCRIPT)
+    result = t.substitute(**mapping)
+    return result
+
+
+def get_command_using_train_func(
+    runtime: types.Runtime,
+    train_func: Callable,
+    train_func_parameters: Optional[Dict[str, Any]],
+    venv_dir: str,
+    train_job_name: str,
+) -> str:
+    """
+    Get the Trainer container command from the given training function and parameters.
+    """
+    # Check if the runtime has a Trainer.
+    if not runtime.trainer:
+        raise ValueError(f"Runtime must have a trainer: {runtime}")
+
+    # Check if training function is callable.
+    if not callable(train_func):
+        raise ValueError(
+            f"Training function must be callable, got function type: {type(train_func)}"
+        )
+
+    # Extract the function implementation.
+    func_code = inspect.getsource(train_func)
+
+    # Extract the file name where the function is defined and move it the venv directory.
+    func_file = Path(venv_dir) / local_exec_constants.LOCAL_EXEC_FILENAME.format(train_job_name)
+
+    # Function might be defined in some indented scope (e.g. in another function).
+    # We need to dedent the function code.
+    func_code = textwrap.dedent(func_code)
+
+    # Wrap function code to execute it from the file. For example:
+    # TODO (andreyvelich): Find a better way to run users' scripts.
+    # def train(parameters):
+    #     print('Start Training...')
+    # train({'lr': 0.01})
+    if train_func_parameters is None:
+        func_code = f"{func_code}\n{train_func.__name__}()\n"
+    else:
+        func_code = f"{func_code}\n{train_func.__name__}({train_func_parameters})\n"
+
+    with open(func_file, "w") as f:
+        f.write(func_code)
+    f.close()
+
+    t = Template(local_exec_constants.LOCAL_EXEC_ENTRYPOINT)
+    mapping = {
+        "PARAMETERS": "",  ## Torch Parameters if any
+        "PYENV_LOCATION": venv_dir,
+        "ENTRYPOINT": " ".join(runtime.trainer.command),
+        "FUNC_FILE": func_file,
+    }
+    entrypoint = t.safe_substitute(**mapping)
+
+    return entrypoint
+
+
+def get_cleanup_script(venv_dir: str, cleanup: bool = True) -> str:
+    script = "\n"
+    if not cleanup:
+        return script
+
+    t = Template(local_exec_constants.LOCAL_EXEC_JOB_CLEANUP_SCRIPT)
+    mapping = {
+        "PYENV_LOCATION": venv_dir,
+    }
+    return t.substitute(**mapping)
+
+
+def get_training_job_command(
```

Comment:
Can we move this to the construction of RuntimeLocalTrainer ?
E.g. get_runtime_local_trainer